src/DirectionalLight.cpp

```diff
@@ -21,3 +22,16 @@ std::string DirectionalLight::get_name() const
     std::string const name = typeid(decltype(*this)).name();
     return name.substr(6);
 }
+
+
+glm::mat4 DirectionalLight::get_projection_view_matrix()
+{
+    if (projection_view_matrix == glm::mat4() && entity != nullptr)
+    {
+        projection_matrix = glm::ortho(-100.0f, 100.0f, -100.0f, 100.0f, 1.0f, 15.0f);
+        view_matrix = glm::lookAt(entity->transform->get_position(), entity->transform->get_position() + entity->transform->get_forward(), glm::vec3(0.0f, 1.0f, 0.0f));
+        projection_view_matrix = projection_matrix * view_matrix;
+    }
+
+    return projection_view_matrix;
```

Comment:
This basically means that we only compute this once, even if the position of directional light changes (which might be helpful when debugging or searching for a good looking position), I'm pretty sure.
We can just cache transform's model matrix and check if it changed.
```cpp
if (m_last_model_matrix != entity->transform->get_model_matrix() && entity != nullptr)
{
    m_last_model_matrix = entity->transform->get_model_matrix();

    projection_matrix = glm::ortho(-100.0f, 100.0f, -100.0f, 100.0f, 1.0f, 15.0f);
    view_matrix = glm::lookAt(entity->transform->get_position(), entity->transform->get_position() + entity->transform->get_forward(), glm::vec3(0.0f, 1.0f, 0.0f));
    projection_view_matrix = projection_matrix * view_matrix;
}
```